.github/workflows/main.yml

```diff
@@ -72,3 +70,87 @@ jobs:
         run: |
           ./devel/lint.sh  $( find . -name 'Dockerfile' \
           -o -name 'Dockerfile.*' )
+
+  build:
+    name: Build stage
+    needs: [lint-golang, lint-extra-files]
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Go
+        uses: actions/setup-go@master
+        with:
+          go-version: 1.14
+        id: go
+
+      - name: Install dependencies
+        env:
+          # yamllint disable-line rule:line-length
+          INSTALL_KUSTOMIZE_URL: "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh"
+          # yamllint disable-line rule:line-length
+          OPERATOR_SDK_URL: "https://github.com/operator-framework/operator-sdk/releases/download/v1.0.0/operator-sdk-v1.0.0-x86_64-linux-gnu"
+          GO111MODULE: "on"
+        run: |
+          # GOPATH="$PWD/go"
+          # GOBIN="$GOPATH/bin"
+          # PATH="$GOBIN:$PATH"
+          curl -s "${INSTALL_KUSTOMIZE_URL}" | bash
+          go install -i
+          go get golang.org/x/lint/golint
+          go get github.com/kisielk/errcheck
+          go get github.com/golangci/golangci-lint/cmd/golangci-lint@v1.30.0
+          # curl --silent -L -o operator-sdk "${OPERATOR_SDK_URL}"
+          # chmod +x operator-sdk
+          # sudo mv operator-sdk /bin/
+
+      # https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow
+      - name: Build
+        env:
+          GO111MODULE: "on"
+          IMG_BASE: "${{ secrets.IMG_BASE }}"
+          DOCKER_AUTH: "${{ secrets.DOCKER_AUTH }}"
+        run: |
+          export PATH=$PATH:$(go env GOPATH)/bin
+          export TAG="${GITHUB_SHA}"
+          if [ "$GITHUB_EVENT_NAME" == "pull_request" ]
+          then
+            export IMG_BASE="local/freeipa-operator"
+          fi
+          export IMG="${IMG_BASE}:${TAG}"
+
+          echo ">>> Building operator binary"
+          make generate
+          go build -o bin/manager main.go
+
+          echo ">>> Building container image"
+          docker build . -t ${IMG}
+          make docker-save
+
+          if [ "$GITHUB_EVENT_NAME" != "pull_request" ]
+          then
+            echo ">>> Pushing container image"
+            DOCKER_REGISTRY="${IMG_BASE%%/*}"
+            [ -e "${HOME}/.docker" ] || mkdir -p "${HOME}/.docker"
+            cat > "$HOME/.docker/config.json" <<< "${DOCKER_AUTH}"
+            docker login "${DOCKER_REGISTRY}"
+            docker push "${IMG}" \
+            && rm -vf docker-image-freeipa-operator.tar.gz
+          fi
```

Comment:
I think this should have some commentary.  To my reading, we only try to push the image for pushes to master, not pull requests.  This makes sense, but a comment will help readers to comprehend the intent.

Reply:
Right. I have commented the block for clarification. Even myself barely remembered the reason when I have reviewed. Good point, thanks.